src/runtime_src/driver/xclng/drm/xocl/userpf/xocl_drv.c

```diff
@@ -165,7 +188,17 @@ int xocl_hot_reset(struct xocl_dev *xdev, bool force)
 	return ret;
 }
 
-void xocl_mb_connect(struct xocl_dev *xdev)
+static void xocl_reset_work(struct work_struct *work)
+{
+	struct xocl_dev *xdev = container_of(to_delayed_work(work),
+			struct xocl_dev, core.reset_work);
+
+	xocl_drvinst_offline(xdev, true);
```

Comment:
this is done in xocl_hot_reset_ioctl() already, why do we set it again here?

Reply:
I thought xocl_reset_work could be called somewhere else in the future. E.g. if xocl_reset_work is called from sysfs, the caller does not have to call offline first. And it is not hurt to call offline twice.